services/121-service/src/transaction-job-processors/transaction-job-processors.service.ts

```diff
@@ -0,0 +1,420 @@
+        });
+        return;
+      } else {
+        throw error;
+      }
+    }
+    // If the transactions was succesful
+
+    let messageType;
+    if (intersolveVisaDoTransferOrIssueCardReturnDto.isNewCardCreated) {
+      messageType = ProgramNotificationEnum.visaDebitCardCreated;
+    } else {
+      messageType = ProgramNotificationEnum.visaLoad;
+    }
+    await this.createMessageAndAddToQueue({
+      type: messageType,
+      programId: input.programId,
+      registration: registration,
+      amountTransferred:
+        intersolveVisaDoTransferOrIssueCardReturnDto.amountTransferredInMajorUnit,
+      bulkSize: input.bulkSize,
+      userId: input.userId,
+    });
+
+    await this.createTransactionAndUpdateRegistration({
+      programId: input.programId,
+      paymentNumber: input.paymentNumber,
+      userId: input.userId,
+      calculatedTranserAmountInMajorUnit:
+        intersolveVisaDoTransferOrIssueCardReturnDto.amountTransferredInMajorUnit,
+      financialServiceProviderId: financialServiceProvider.id,
+      registration,
+      oldRegistration,
+      isRetry: input.isRetry,
+      status: StatusEnum.success,
+    });
+  }
+
+  private async createTransactionAndUpdateRegistration({
+    programId,
+    paymentNumber,
+    userId,
+    calculatedTranserAmountInMajorUnit,
+    financialServiceProviderId,
+    registration,
+    oldRegistration,
+    isRetry,
+    status,
+    errorText: errorMessage,
+  }: ProcessTransactionResultInput): Promise<void> {
+    const resultTransaction = await this.createTransaction({
+      amount: calculatedTranserAmountInMajorUnit,
+      registration: registration,
+      financialServiceProviderId: financialServiceProviderId,
+      programId: programId,
+      paymentNumber: paymentNumber,
+      userId: userId,
+      status: status,
+      errorMessage: errorMessage,
+    });
+
+    await this.latestTransactionRepository.insertOrUpdateFromTransaction(
+      resultTransaction,
+    );
+
+    if (!isRetry) {
+      await this.updatePaymentCountAndStatusInRegistration(
+        registration,
+        programId,
+      );
+      // Added this check to avoid a bit of processing time if the status is the same
+      if (
+        oldRegistration.registrationStatus !== registration.registrationStatus
+      ) {
+        await this.eventsService.log(
+          {
+            id: oldRegistration.id,
+            status: oldRegistration.registrationStatus ?? undefined,
+          },
+          {
+            id: registration.id,
+            status: registration.registrationStatus ?? undefined,
+          },
+          {
+            registrationAttributes: ['status'],
+          },
+        );
+      }
+    }
+  }
+
+  private async addMessageJobToQueue({
+    registration,
+    userId,
+    message,
+    bulksize,
+  }: {
+    registration: RegistrationEntity | Omit<RegistrationViewEntity, 'data'>;
+    userId: number;
+    message?: string;
+    messageTemplateKey?: string;
```

Comment:
```suggestion
```

You aren't using this.

Reply:
Check if this is a bug: see function params deconstruction for what is missing.

Reply:
Fixed in new branch.